pcsx2/Sio.cpp

```diff
@@ -120,13 +120,26 @@ void sioInit()
 
 bool isR3000ATest = false;
 
+u16 currentStat = 0, lastStat = 0, currentMode = 0, lastMode = 0, currentBaud = 0, lastBaud = 0;
+
 // Check the active game's type, and fire the matching interrupt.
 // The 3rd bit of the HW_IFCG register lets us know if PSX mode is active. 1 = PSX, 0 = PS2
 // Note that the R3000A's call to interrupts only calls the PS2 based (lack of) delays.
 SIO_FORCEINLINE void sioInterrupt() {
 	if ((psxHu32(HW_ICFG) & (1 << 3)) && !isR3000ATest) {
-		if (!(psxRegs.interrupt & (1 << IopEvt_SIO)))
-			PSX_INT(IopEvt_SIO, 64); // PSXCLK/250000);
+		if (!(psxRegs.interrupt & (1 << IopEvt_SIO))) {
+			currentStat = sio.StatReg & 67106816; // Mask out bits 11-25
+			currentMode = sio.ModeReg & 3; // Mask out bits 0-1
+			currentBaud = sio.BaudReg; // The entire baud reg
+			// If anything changes, print.
+			if (currentStat != lastStat || currentMode != lastMode || currentBaud != lastBaud)
```

Comment:
I'm not sure how likely it is to get 0 value on all of these bitfields, if such a scenario arises then the message won't be printed. Maybe ``-1`` could be a better default value (of course also changing it to signed in that case).